pkg/apis/maistra/conversion/discovery_selectors.go

```diff
@@ -0,0 +1,51 @@
+package conversion
+
+import (
+	v1 "github.com/maistra/istio-operator/pkg/apis/maistra/v1"
+	v2 "github.com/maistra/istio-operator/pkg/apis/maistra/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func populateDiscoverySelectorsValues(in *v2.ControlPlaneSpec, out map[string]interface{}) error {
+	if in.MeshConfig == nil || in.MeshConfig.DiscoverySelectors == nil {
+		return nil
+	}
+
+	if len(in.MeshConfig.DiscoverySelectors) > 0 {
```

Comment:
You need to remove this if statement (just this line, not the entire if block) to make the `empty` test pass.